module/zfs/dnode.c

```diff
@@ -1778,15 +1778,24 @@ dnode_try_claim(objset_t *os, uint64_t object, int slots)
 }
 
 /*
- * Checks if the dnode contains any uncommitted dirty records.
+ * Checks if the dnode itself is dirty, or is carrying any uncommitted records.
+ * Its important to check both, as some operations (eg appending to a file) can
```

Comment:
Its important to check both, as some operations (eg appending to a file) --->
It is important to check both conditions, as some operations (e.g. appending to a file)

Reply:
I think it's fine without the word "condition", but yes:
```suggestion
 * Its -> It's
```

Reply:
It is indeed fine without "condition" here but if you read the comment as a whole, you would find both being used several times while referring to different things. The suggestions are therefore to be taken as a whole.

Reply:
Yeah, its a good clarifier. Taken, thanks!